tests/lib/hardware/openstack_libcloud.py

```diff
@@ -78,19 +79,27 @@ def boot(self):
         )
         logging.debug(f"node {self.name} booted")
 
-        self._create_and_attach_floating_ip()
+        self._get_floating_ip()
         # Wait for node to be ready
         self._wait_until_state(NodeState.RUNNING)
         # Attach a 10GB disk
         self._create_and_attach_volume(10)
 
-    def _create_and_attach_floating_ip(self):
-        # TODO(jhesketh): Move cloud-specific configuration elsewhere
-        floating_ip = self.conn.ex_create_floating_ip(
-            config.OS_EXTERNAL_NETWORK)
-
-        logger.info(f"Created floating IP: {floating_ip}")
-        self._floating_ips.append(floating_ip)
+    def _get_floating_ip(self):
+        try:
+            floating_ip = self.conn.ex_create_floating_ip(
+                config.OS_EXTERNAL_NETWORK)
+            logger.info("Created floating IP: ")
+            logger.info(floating_ip)
+            self._floating_ips.append(floating_ip)
+            self._floating_ips_created.append(floating_ip)
```

Comment:
I wonder if a single list with a tuple or dict would be better. Like `self._floating_ips.append((floating_ip, True))` where `True` indicates that this ip was created. What do you think?

Reply:
I'm not sure this makes it clearer.
No strong opinion actually.

Reply:
ok. let's keep it.